improver/blending/blend_across_adjacent_points.py

```diff
@@ -155,11 +150,15 @@ def process(self, cube):
 
         # Calculate weights and produce blended output.
         weights = self.WeightsPlugin(cube, self.coord, self.central_point)
-        blended_cube = self.BlendingPlugin(cube, weights)
 
-        # With one threshold dimension (such as for low cloud), the threshold
-        # axis is demoted to a scalar co-ordinate by BlendingPlugin. This line
-        # promotes threshold to match the dimensions of central_point_cube.
-        blended_cube = check_cube_coordinates(central_point_cube, blended_cube)
-        blended_cube = central_point_cube.copy(blended_cube.data)
+        dim_to_collapse = cube.coord_dims(self.coord)[0]
+        indices = [slice(None)] * cube.ndim
+        # Create the cube using the same metadata as the central cube, fill it with zeros
+        blended_cube = central_point_cube.copy(np.zeros_like(central_point_cube.data))
+
+        number_of_subcubes = cube.shape[dim_to_collapse]
+        for subcube_index in range(number_of_subcubes):
+            indices[dim_to_collapse] = subcube_index
+            blended_cube.data += cube[tuple(indices)].data * weights.data[subcube_index]
+
```

Comment:
How much benefit does this provide on top of the slicing added in the blending plugin? The disadvantage of this is that we loose the common code path for blending.